screens/UserScreens/UserProfile.jsx

```diff
@@ -31,20 +30,82 @@ export default function UserProfileScreen({ navigation }) {
 
     useEffect(() => {
         get(child(dbRef(db), `users/${auth.currentUser.uid}`)).then((snapshot) => {
-            setUserName(snapshot.exists() ? [snapshot.val().firstName, snapshot.val().lastName] : "Anonymous");
+            setUserName(
+                snapshot.exists()
+                    ? [snapshot.val().firstName, snapshot.val().lastName]
+                    : "Anonymous"
+            );
         });
     }, []);
 
     const handleSignOut = () => {
         try {
             signOut(auth);
-        } catch (error) { 
+        } catch (error) {
             alert("sign out failed:", error.message);
         }
     };
 
+    // TODO: Check this works w multiple admins!
     const handleDeleteAccount = () => {
-        alert("Delete Account");
+        try {
+            // Find all of user's productions
+            get(child(dbRef(db), `users/${auth.currentUser.uid}/productions`)).then((snapshot) => {
+                if (!snapshot.exists()) {
+                    return;
+                }
+                const productions = snapshot.val();
+                // Delete user from all productions
+                Object.keys(productions).forEach((playCode) => {
+                    remove(
+                        dbRef(db, `productions/${playCode}/participants/${auth.currentUser.uid}`)
+                    );
+                    get(child(dbRef(db), `productions/${playCode}/admins`)).then((snapshot) => {
+                        if (!snapshot.exists()) {
+                            return;
+                        }
+                        const admins = Object.keys(snapshot.val());
+                        if (admins.includes(auth.currentUser.uid)) {
+                            if (admins.length === 1) {
+                                // If only one admin, add someone in from participants or delete production
+                                get(child(dbRef(db), `productions/${playCode}/participants`)).then(
+                                    (snapshot) => {
+                                        if (snapshot.exists()) {
+                                            const participants = Object.keys(snapshot.val());
+                                            // If there are other participants, make first one an admin
+                                            set(dbRef(db, `productions/${playCode}/admins`), {
+                                                [participants[0]]: Date.now(),
+                                            });
+                                        } else {
+                                            // If there are no participants, delete production.
```

Comment:
probably eventually have some sort of pop up like "you are the only member of this production, so this will delete the whole production. continue?" but overall functionality and readability is all there so im happy to push.

Reply:
Idk if it's needed icl cos it shouldn't affect any other user? The existing user should know from the fact they're deleting their account that their role in productions is lost